source/server/guarddog_impl.cc

```diff
@@ -105,7 +103,14 @@ void GuardDogImpl::step() {
                  static_cast<size_t>(ceil(multi_kill_fraction_ * watched_dogs_.size())));
 
     for (auto& watched_dog : watched_dogs_) {
-      const auto last_checkin = watched_dog->dog_->lastTouchTime();
+      if (watched_dog->dog_->touchCount() > 0) {
+        // Watchdog was touched since the guard dog last checked; update time and reset the count.
+        watched_dog->dog_->resetTouchCount();
```

Comment:
perf nit: I think unless you do a relaxed load it's probably faster to just always do a reset and return the previous count, and then branch on that. Up to you.

Reply:
I changed the watchdog API so it provides a single method that returns if the dog was touched recently and reset touch state.  I also changed the type to bool since it's more appropriate and switched to a looser memory order for performance.